### ISSUE TITLE
Unit and Description of OBD signals

### ISSUE THREAD

**Issue by erikbosch** [GENIVI/vehicle_signal_specification#250]
When reviewing the proposed Android to VSS mapping ( https://at.projects.genivi.org/jira/browse/AASIG-122) i realized that our OBD.vspec is a bit odd. According to https://en.wikipedia.org/wiki/OBD-II_PIDs there are a lot of conversions going on. If we take the two examples below

- EngineSpeed (0C) is sent as uint16, to get the "real value" take (256*A + B)/4
- OilTemperature (5C) is sent as uint8, to get the "real value" take A - 40

And this is not obvious in our current descriptions: 

```
- EngineSpeed:
  datatype: float
  type: sensor
  unit: rpm
  description: PID 0C - Engine speed measured as rotations per minute

- OilTemperature:
  datatype: uint8
  type: sensor
  unit: celsius
  description: PID 5C - Engine oil temperature
```

For EngineSpeed we state what the real value can be (a float, but actually not all floats as we just can support quarters). For OilTemperature we state what is sent (a uint8). Just from looking at OilTemperature above it is not obvious that you can send negative temperatures.  

I can see two approaches for OBD-signals:

**Represent the real value**

I.e. do not care how OBD send it. With this approach 

- OilTemperature shall be an int16 with min: -40 and max +210. Unit shall be celsius.
- EngineSpeed shall be a float with min: 0 and max: 16,383.75. Unit shall be rpm. 

**Represent the value transmitted by OBD**

I.e. do not care what value it represent. With this approach 

- OilTemperature shall be a uint8. 
- EngineSpeed shall be a uint16. 

With this approach I suggest that celsius/rpm shall be removed from the signals. As of now VSS does not have any syntax for scaling/offset. So if we say "celsius+uint8" I assume that I do not need to do any scaling/offset. I.e. in practice - OBD is using a custom unit - not the unit "celsius" standardized by VSS. Of course we can include the conversion as a comment or in the description, and mention celsius in a comment.

As of now it seems that we have a mix of the two approaches. I believe this needs to be corrected!

**Comment by danielwilms** [comment]
09/03/21 intention is to use the real value

**Comment by erikbosch** [comment]
I did a quick review of all OBD values in OBD.vspec. To me it seems that it is quite inconsistent:

- Often we use uint8 (e.g. for percent) even if OBD supports fractions (e.g. 253/255)
- Sometimes we use float even if OBD only supports integer values
- Sometimes we use uint8 even if OBD supports negative values
- Sometimes we use e.g. uint32 even if uint16 would suffice

For some identified issues see https://github.com/erikbosch/vehicle_signal_specification/pull/2/files (where I also added ranges specified by OBD). There are also some concerns on how e.g. strings shall be used. I could for instance not understand the VSS representation of PID 41.

I can see some different approaches going forward:

- Change to "strict" VSS-representation everywhere , i.e. use float only if decimals/fractions needs to be supported, select right int-type depending on supported range in OBD and so on. With this it is easy that we do mistakes.
- Ignore OBD-limits and representation/scaling/offset. Always use float in OBD.vspec (unless it is an enum/bitmask or similar). 
- Wait with doing anything until we see someone with urgent needs (i.e. lets us consider OBD as of now to be "work in progress")
